Content.Goobstation.Shared/Body/InsideBodyPartSystem.cs

```diff
@@ -0,0 +1,98 @@
+using Content.Goobstation.Common.Body;
+using Content.Shared.Actions;
+using Content.Shared.Body.Components;
+using Content.Shared.Body.Part;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Damage;
+using Content.Shared.DoAfter;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Jittering;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+
+namespace Content.Goobstation.Shared.Body;
+
+public sealed class InsideBodyPartSystem : CommonInsideBodyPartSystem
+{
+    [Dependency] private readonly DamageableSystem _damage = default!;
+    [Dependency] private readonly ItemSlotsSystem _slots = default!;
+    [Dependency] private readonly SharedActionsSystem _actions = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedJitteringSystem _jittering = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly MobStateSystem _mob = default!;
+
+    private EntityQuery<BodyComponent> _bodyQuery;
+    private EntityQuery<BodyPartComponent> _partQuery;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _bodyQuery = GetEntityQuery<BodyComponent>();
+        _partQuery = GetEntityQuery<BodyPartComponent>();
+
+        SubscribeLocalEvent<InsideBodyPartComponent, BodyPartBurstEvent>(OnAction);
+        SubscribeLocalEvent<InsideBodyPartComponent, BurstDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<InsideBodyPartComponent, ComponentShutdown>(OnShutdown);
+    }
+
+    public override void InsertedIntoPart(EntityUid item, EntityUid part)
+    {
+        var comp = EnsureComp<InsideBodyPartComponent>(item);
+        comp.Part = part;
+        Dirty(item, comp);
+        if (!_bodyQuery.HasComp(item))
+            return;
+
+        _actions.AddAction(item, ref comp.ActionEntity, comp.BurstAction);
+        if (comp.ActionEntity is {} action)
+            _actions.SetEntityIcon(action, part);
+    }
+
+    public override void RemovedFromPart(EntityUid item)
+    {
+        RemComp<InsideBodyPartComponent>(item);
+    }
```

Comment:
```suggestion
    public override void RemovedFromPart(EntityUid item) =>
        RemComp<InsideBodyPartComponent>(item);
```
<img width="235" height="214" alt="image" src="https://github.com/user-attachments/assets/4c3a969f-d9b8-4218-8f30-2b9ffc3b378c" />